pkg/sources/github/feedlet.go

```diff
@@ -0,0 +1,396 @@
+
+	service := resources.MakeService(t.feedNamespace, serviceName, t.image, t.feedServiceAccountName, target, trigger.Parameters[secretName].(string), trigger.Parameters[secretKey].(string))
+	service, createErr := sc.Create(service)
+	if createErr != nil {
+		glog.Errorf("service failed: %s", createErr)
+	}
+
+	return service, createErr
+}
+
+func (t *GithubEventSource) deleteReceiveAdapter(namespace string, serviceName string) error {
+	sc := t.servingclientset.ServingV1alpha1().Services(t.feedNamespace)
+
+	// First, check if deployment exists already.
+	if _, err := sc.Get(serviceName, metav1.GetOptions{}); err != nil {
+		if !apierrs.IsNotFound(err) {
+			glog.Infof("services.Get for %q failed: %s", serviceName, err)
+			return err
+		}
+		glog.Infof("service %q already deleted", serviceName)
+		return nil
+	}
+
+	return sc.Delete(serviceName, &metav1.DeleteOptions{})
+}
+
+
+func (t *GithubEventSource) deleteWebhook(trigger sources.EventTrigger, feedContext sources.FeedContext) error {
+	glog.Infof("Stopping github webhook feed with context %+v", feedContext)
+
+	components := strings.Split(trigger.Resource, "/")
+	owner := components[0]
+	repo := components[1]
+
+	if _, ok := feedContext.Context[webhookIDKey]; !ok {
+		// there's no webhook id, nothing to do.
+		glog.Infof("No Webhook ID Found, bailing...")
+		return nil
+	}
+	webhookID := feedContext.Context[webhookIDKey].(string)
+
+	ctx := context.Background()
+	ts := oauth2.StaticTokenSource(
+		&oauth2.Token{AccessToken: trigger.Parameters["accessToken"].(string)},
+	)
+	tc := oauth2.NewClient(ctx, ts)
+	client := ghclient.NewClient(tc)
+
+	id, err := strconv.ParseInt(webhookID, 10, 64)
+	if err != nil {
+		glog.Warningf("Failed to convert webhook %q to int64 : %s", webhookID, err)
+		return err
+	}
+	r, err := client.Repositories.DeleteHook(ctx, owner, repo, id)
+	if err != nil {
+		if errResp, ok := err.(*ghclient.ErrorResponse); ok {
+			// If the webhook doesn't exist, nothing to do
+			if errResp.Message == "Not Found" {
+				glog.Info("Webhook doesn't exist, nothing to delete.")
+				return nil
+			}
+		}
+		glog.Warningf("Failed to delete the webhook: %s", err)
+		glog.Warningf("Response:\n%+v", r)
+		return err
+	}
+	glog.Infof("Deleted webhook %q successfully", webhookID)
+	return nil
+}
+
+func (t *GithubEventSource) createWebhook(trigger sources.EventTrigger, domain string) (*sources.FeedContext, error) {
+
+	glog.Infof("CREATING GITHUB WEBHOOK")
+
+	ctx := context.Background()
+	ts := oauth2.StaticTokenSource(
+		&oauth2.Token{AccessToken: trigger.Parameters[accessToken].(string)},
+	)
+	tc := oauth2.NewClient(ctx, ts)
+
+	client := ghclient.NewClient(tc)
+	active := true
+	name := "web"
+	config := make(map[string]interface{})
+	config["url"] = fmt.Sprintf("http://%s", domain)
+	config["content_type"] = "json"
+	config["secret"] = trigger.Parameters[secretToken].(string)
+	hook := ghclient.Hook{
+		Name:   &name,
+		URL:    &domain,
+		Events: []string{"pull_request"},
```

Comment:
This should be dynamic. We're going to need a map from our CloudEvents types to their native types. Feel free to do a TODO for now I guess or use the trigger.EventType.